core/ArchiveProcessor/Loader.php

```diff
@@ -119,20 +123,77 @@ private function prepareArchiveImpl($pluginName)
         // invalidate existing archives before we start archiving in case data was tracked in the past. if the archive is
         // made invalid, we will correctly re-archive below.
         if ($this->invalidateBeforeArchiving
-            && Rules::isBrowserTriggerEnabled()
+          && Rules::isBrowserTriggerEnabled()
         ) {
             $this->invalidatedReportsIfNeeded();
         }
+        // load existing data from archive
+        list($visits, $visitsConverted) = $this->loadArchiveData();
+        if ($this->quickReturn) {
+            return [$visits, $visitsConverted];
+        }
+
+        if (SettingsServer::isArchivePhpTriggered()) {
+            $this->logger->info("initiating archiving via core:archive for " . $this->params);
+        }
+
+        //get Lock ID
+        $lockId = $this->makeArchivingLock();
 
+        //ini lock
+        $this->lock = new LoaderLock($lockId);
```

Comment:
If this is only used in that method, there is actually no need in having a class property for that and a local variable can be used.